src/main/java/jackyy/dimensionaledibles/block/BlockCustomCake.java

```diff
@@ -1,143 +1,118 @@
+                               s + " is not a valid input line! Format needs to be: <dimID>, <x>, <y>, <z>");
                     continue;
                 }
-                if (Integer.parseInt(parts[0].trim()) == dimension) {
+                if (Integer.parseInt(parts[0].trim()) == cakeDimension) {
                     customX = Integer.parseInt(parts[1].trim());
                     customY = Integer.parseInt(parts[2].trim());
                     customZ = Integer.parseInt(parts[3].trim());
                 }
-            } catch (NumberFormatException e) {
-                DimensionalEdibles.logger.log(Level.ERROR, s + " is not a valid line input! The dimension ID needs to be a number!");
-            }
-        }
-        if (!stack.isEmpty() && stack.getItem() == Item.REGISTRY.getObject(new ResourceLocation(fuel))) {
-            if (meta >= 0) {
-                world.setBlockState(pos, state.withProperty(BITES, meta), 2);
-                if (!player.capabilities.isCreativeMode) {
-                    stack.shrink(1);
-                }
-                return true;
-            }
-        } else {
-            if (world.provider.getDimension() != dimension) {
-                if (!world.isRemote) {
-                    if (player.capabilities.isCreativeMode || !ModConfig.tweaks.customEdible.customCake.consumeFuel) {
-                        teleportPlayer(world, player, dimension);
-                    } else {
-                        consumeCake(world, pos, player, dimension);
-                    }
-                    return true;
-                }
+            } catch(NumberFormatException e) {
+                logger.log(Level.ERROR,
+                           s + " is not a valid line input! The dimension ID needs to be a number!");
             }
         }
-        return false;
     }
 
-    private void teleportPlayer(World world, EntityPlayer player, int dimension) {
-        EntityPlayerMP playerMP = (EntityPlayerMP) player;
-        BlockPos coords;
-        if (customX != 0 && customY != 0 && customZ != 0) {
-            coords = new BlockPos(customX, customY, customZ);
-        } else {
-            coords = TeleporterHandler.getDimPos(playerMP, dimension, player.getPosition());
-        }
-        TeleporterHandler.updateDimPos(playerMP, world.provider.getDimension(), player.getPosition());
-        TeleporterHandler.teleport(playerMP, dimension, coords.getX(), coords.getY(), coords.getZ(), playerMP.server.getPlayerList());
-    }
-
-    private void consumeCake(World world, BlockPos pos, EntityPlayer player, int dimension) {
-        if (player.canEat(true)) {
-            int l = world.getBlockState(pos).getValue(BITES);
-            if (l < 6) {
-                player.getFoodStats().addStats(2, 0.1F);
-                world.setBlockState(pos, world.getBlockState(pos).withProperty(BITES, l + 1), 3);
-                teleportPlayer(world, player, dimension);
+    private String determineCakeFuel() {
```

Comment:
This could be parsed on config change.

Reply:
That sounds like a sensible optimization to make, but I think such changes should be implemented separately. The goal of this refactoring was to not change any of the functionality but rather get rid of copy-pasted code.